packages/ragbits-core/src/ragbits/core/cli.py

```diff
@@ -1,104 +1,15 @@
-
-    return prompt_cls()
-
-
-class LLMResponseCliOutput(BaseModel):
-    """An output model for llm responses in CLI"""
-
-    question: ChatFormat
-    answer: str | BaseModel | None = None
-
 
-prompts_app = typer.Typer(no_args_is_help=True)
+from ragbits.core.prompt._cli import prompts_app
+from ragbits.core.vector_stores._cli import vector_stores_app
 
 
-def register(app: CLI) -> None:
+def register(app: typer.Typer) -> None:
     """
     Register the CLI commands for the package.
 
     Args:
         app: The Typer object to register the commands with.
     """
-
-    @prompts_app.command()
-    def lab(
-        file_pattern: str = core_config.prompt_path_pattern,
-        llm_factory: str = core_config.default_llm_factories[LLMType.TEXT],
-    ) -> None:
-        """
-        Launches the interactive application for listing, rendering, and testing prompts
-        defined within the current project.
-        """
-        from ragbits.core.prompt.lab.app import lab_app
-
-        lab_app(file_pattern=file_pattern, llm_factory=llm_factory)
-
-    @prompts_app.command()
-    def generate_promptfoo_configs(
-        file_pattern: str = core_config.prompt_path_pattern,
-        root_path: Path = Path.cwd(),  # noqa: B008
-        target_path: Path = Path("promptfooconfigs"),
-    ) -> None:
-        """
-        Generates the configuration files for the PromptFoo prompts.
-        """
-        from ragbits.core.prompt.promptfoo import generate_configs
-
-        generate_configs(file_pattern=file_pattern, root_path=root_path, target_path=target_path)
-
-    @prompts_app.command()
-    def render(prompt_path: str, payload: str | None = None) -> None:
-        """
-        Renders a prompt by loading a class from a module and initializing it with a given payload.
-        """
-        prompt = _render(prompt_path=prompt_path, payload=payload)
-        response = LLMResponseCliOutput(question=prompt.chat)
-        app.print_output(response)
-
-    @prompts_app.command(name="exec")
-    def execute(
-        prompt_path: str,
-        payload: str | None = None,
-        llm_factory: str | None = core_config.default_llm_factories[LLMType.TEXT],
-    ) -> None:
-        """
-        Executes a prompt using the specified prompt class and LLM factory.
-
-        Raises:
-            ValueError: If `llm_factory` is not provided.
-        """
-        prompt = _render(prompt_path=prompt_path, payload=payload)
-
-        if llm_factory is None:
-            raise ValueError("`llm_factory` must be provided")
-        llm: LLM = LLM.subclass_from_factory(llm_factory)
-
-        llm_output = asyncio.run(llm.generate(prompt))
-        response = LLMResponseCliOutput(question=prompt.chat, answer=llm_output)
-        app.print_output(response)
-
     app.add_typer(prompts_app, name="prompts", help="Commands for managing prompts")
+    app.add_typer(vector_stores_app, name="vector-store", help="Commands for managing vector stores")
```

Comment:
i thought that maybe for the given module we should also auto-register sub module's cli's - however doesn't seem as sth very important

Reply:
Personally, I wouldn't. We need autodiscovery on the package level because packages cannot rely on CLI being installed (since they don't depend on it). There is no similar problem on the inter-package level, so I think explicit (and fast) is better in this case than implicit and slower.